.tekton/tasks/merge-pr.yaml

```diff
@@ -0,0 +1,117 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: merge-pr
+spec:
+  description: |
+    merge the pull request into main branch.
+  params:
+    - name: pull_request_number
+    - name: TARGET_GH_OWNER
+      description: Owner of GitHub repository of the infra-deployments code
+      default: redhat-appstudio
+    - name: TARGET_GH_REPO
+      description: GitHub repository of the infra-deployments code
+      default: infra-deployments
+    - name: source_branch
+  workspaces:
+   - name: shared-file-path
+  steps:
+    - name: merge-pull-request
+      image: quay.io/redhat-appstudio/github-app-token@sha256:b4f2af12e9beea68055995ccdbdb86cfe1be97688c618117e5da2243dc1da18e
+      workingDir: $(workspaces.shared-file-path.path)
+      env:
+        - name: GITHUB_GRAPHQL_API_URL
+          value: https://api.github.com/graphql
+        - name: PULL_REQUEST_NUMBER
+          value: $(params.pull_request_number)
+        - name: SOURCE_BRANCH
+          value: $(params.source_branch)
+        - name: TARGET_GH_OWNER
+          value: "$(params.TARGET_GH_OWNER)"
+        - name: TARGET_GH_REPO
+          value: "$(params.TARGET_GH_REPO)"
+      script: |
+        #!/usr/bin/env python3
```

Comment:
Why not `git push -u origin HEAD:main`?

Reply:
@Roming22 I tried using this approach but everytime getting error:
```
git push origin HEAD:origin/main
fatal: could not read Username for 'https://github.com': No such device or address
```

I also added the required global config and token but still getting the same error. So for now I have separated out the code in the `.tekton/hack` folder and using it in the task